tests/integration/templates/deploy_template_3.yaml

```diff
@@ -50,8 +50,8 @@ Resources:
       - AttributeName: startTime
         KeyType: RANGE
       ProvisionedThroughput:
-        ReadCapacityUnits: '5'
```

Comment:
Btw - just a note here. We sometimes added "invalid" data types to the tests on purpose. 

Reason: in quite a few cases the schema in the AWS CF docs is documented to be more strict than it actually is in reality. For example, if numbers are required as per the schema, number strings are also often ok in real AWS. 

Would be good to take this into account, and make sure we have those cases covered as well (to make sure we have proper conversion logic in place).

Maybe change this to:
```
      ProvisionedThroughput:
        ReadCapacityUnits: '5'
        WriteCapacityUnits: 5
```
?

Reply:
This was actually leading to flaky tests before. If I remember correctly this test also *sometimes* failed in our CI environment.

If we identify things where CloudFormation is rather non-strict about this, we should definitely do this on a more per-attribute basis (and have corresponding tests for AWS to back this up).

Reply:
Guess for now the best course of action is to either leave it so that the tests are working properly or make the casting a bit smarter. :thinking: 

In general though this will be something that is easier (and more permanently) fixed when we start using the cloudformation schema definitions since they also include the parameter types, so I'm not sure if making the casting better and enhancing the retry is worth the effort.

Reply:
Good point, agreed. Will be part of a future, more holistic refactoring.. 👍